nexus_constructor/main_window_utils.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+from PySide2.QtCore import QModelIndex
+from PySide2.QtGui import QIcon
+from PySide2.QtWidgets import QAction, QToolBar, QWidget, QTreeView
+
+from nexus_constructor.component.component import Component
+from nexus_constructor.component.link_transformation import LinkTransformation
+from nexus_constructor.component.transformations_list import TransformationsList
+from nexus_constructor.component_tree_model import ComponentTreeModel
+from nexus_constructor.transformations import Transformation
+
+
+def create_and_add_toolbar_action(
+    icon_path: str,
+    mouse_over_text: str,
+    trigger_method: classmethod,
+    component_tool_bar: QToolBar,
+    component_tree_view_tab: QWidget,
+    set_enabled: bool = False,
+) -> QAction:
+    """
+    Create a QAction and add it to the component toolbar.
+    :param icon_path: The location of the action icon relative to the "ui" folder.
+    :param mouse_over_text: The text that should appear when the mouse is above the icon.
+    :param trigger_method: The method that should be called when the icon is clicked.
+    :param component_tool_bar: The tool bad that the action is added to.
+    :param component_tree_view_tab: The tab for the component tree view.
+    :param set_enabled: A bool indicating whether or not the action should be enabled immediately after it's been
+        created. Only needs to be true for the Add Component button.
+    :return The new QAction.
+    """
+    toolbar_action = QAction(
+        QIcon(os.path.join("ui", icon_path)), mouse_over_text, component_tree_view_tab
+    )
+    toolbar_action.triggered.connect(trigger_method)
+    toolbar_action.setEnabled(set_enabled)
+    component_tool_bar.addAction(toolbar_action)
+    return toolbar_action
+
+
+def set_button_state(
+    component_tree_view: QTreeView,
+    delete_action: QAction,
+    duplicate_action: QAction,
+    new_rotation_action: QAction,
+    new_translation_action: QAction,
+    create_link_action: QAction,
+    zoom_action: QAction,
+    edit_component_action: QAction,
+):
+    """
+    Changes the button states based on user interaction with the component tree view.
+    :param component_tree_view: The component tree view.
+    :param delete_action: The action for deleting an item.
+    :param duplicate_action: The action for duplicating an item.
+    :param new_rotation_action: The action for creating a new rotation.
+    :param new_translation_action: The action for creating a new translation.
+    :param create_link_action: The action for creating a link.
+    :param zoom_action: The action for zooming on a component.
+    :param edit_component_action: The action for editing a component.
+    """
+    indices = component_tree_view.selectedIndexes()
```

Comment:
I don't think `indices` is a very clear name, especially considering this application deals with 3d shapes. Can you rename it to `selected_items`?